docs/general.md

```diff
@@ -534,4 +534,82 @@ add 8,r0,r1
 
 ## Bootloader
 
-Operator execution (transactions come in, get executed on the bootloader, state is suspended until new transaction shows up).
+The **Bootloader** is a system contract that orchestrates the construction and validation of new blocks, acting as a bridge between the EraVM and the external server to ensure accurate transaction processing.
+
+It validates and executes transactions, handles errors effectively, and integrates with EraVM for seamless block construction.
+
+### Core Functions
+1. **Transaction Lifecycle Management**:
+    - **Validation**: Validates each transaction from the server to meet the required criteria.
+    - **Fee Processing**: Calculates and deducts the necessary transaction fees.
+    - **Execution**: Executes transactions, forming the new block.
+
+    Simplified workflow:
```

Comment:
Can we state a bit more that this is a high-level idea on how the boot loader works? For clarity's sake.

Reply:
Fixed! I added a clarifying line just before this.